tests/unit/s2n_client_supported_groups_extension_test.c

```diff
@@ -260,12 +260,19 @@ int main()
 
             for (size_t i = 0; i < NUM_MISMATCH_PQ_TEST_POLICY_OVERRIDES; i++) {
                 EXPECT_SUCCESS(s2n_enable_tls13_in_test());
+                DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(),
+                        s2n_config_ptr_free);
+                EXPECT_NOT_NULL(config);
+                EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "default_tls13"));
```

Comment:
Nit: Why bother creating a config? Couldn't you just use s2n_connection_set_cipher_preferences?

Reply:
s2n_connection_set_cipher_preferences doesnt work.

`s2n_connection_set_cipher_preferences` sets the `conn->security_policy_override` field but it get overridden by this test which is also directly modifying the field.

Reply:
The connection policy always takes precedence over the config policy. So if the test is overriding the connection policy, why is setting the config policy working?

Reply:
Yea good point. Let me check if I was setting the cipher pref after the testing override behavior.

Reply:
Yeah if we have to do config here instead of connection that's fine, but I'm now worried about WHY we have to do config. I think we should confirm that before merging this.

Reply:
The test fix requires setting the config because the actual test case is setting `client->security_policy_override` to NULL. This means that we end up using `client->config->ecc_prefs` and the ecc_prefs for a FIPS policy doesn't have s2n_ecc_curve_x25519

Here is a detailed explanation of whats happening: https://gist.github.com/toidiu/c3ddeca499ba3bcf83328520b8a18cc4

Reply:
That makes much more sense! Thanks for confirming.

That makes this test kind of tricky though. The test isn't just relying on the policy supporting x25519 and TLS1.3, it's also relying on the policy not supporting pq. And I don't think it'd fail if that assumption isn't met. I wonder if it'd be safer to use a named policy instead of "default_tls13", and to leave a comment explaining why that policy was chosen. I'm imagining if/when we want to someday add pq to the default policy :upside_down_face: 

To be clear, you're not making the test worse by using "default_tls13". The test already was missing safeguard around its policy assumptions.

Reply:
I was considering "default_tls13" deprecated but you can never be certain of that. updated in https://github.com/aws/s2n-tls/pull/4926/commits/371ceec7f819b5f73ebd5a6f891dcd0d48373243